### PR TITLE
add metric filter support to cloudwatch logs

### PR DESCRIPTION
**Pull request by brady-gsa**

### CODE REVIEW

File: moto/logs/metric_filters.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                and metric_name_matches
+                and namespace_matches
+            ):
+                result.append(f)
+
+        return result
+
+    def delete_filter(self, filter_name=None, log_group_name=None):
+        for f in self.metric_filters:
+            if f["filterName"] == filter_name:
```
Comment (bblommers) [inline moto/logs/metric_filters.py]: Combine these if-statements (if x and y)

File: moto/logs/responses.py [hunk 2]
```diff
@@ -7,6 +9,9 @@
 
 
 class LogsResponse(BaseResponse):
+    def __init__(self):
+        super().__init__()
```
Comment (bblommers) [inline moto/logs/responses.py]: Travis isn't too happy about this - it fails in Python 2. Any particular reason you added it?

Comment (brady-gsa) [reply moto/logs/responses.py]: I believe I was using it before re-factoring. Shouldn't be needed anymore.

**Comment by brady-gsa** [comment]
> Hi @brady-gsa, thanks for your PR!
> 
> There's quite a few validation cases - would you mind adding some tests to cover those?

I've added validation test cases

**Comment by bblommers** [comment]
Hey @brady-gsa, Travis is still failing on a few test cases: https://travis-ci.org/github/spulec/moto/builds/668765666
Happy to merge once the build passes

**Comment by bblommers** [comment]
This has now been superceded by #4278

**Pull request closed without merging**